docs/README.md

```diff
@@ -0,0 +1,33 @@
+# ONS Technical Documentation
+
+## Project overview
+
+To work on this project, see the [README](../README.md).
+
+## Table of contents
+
+- Conventions
+    - [Type hinting](conventions/type_hinting.md)
+- Integrations
+    - [Authentication service](integrations/auth.md)
+    - [Dataset API](integrations/dataset_api.md)
+    - [Search service](integrations/search_service.md)
```

Comment:
This section doesn't look like the whole table of contents, should it have a different header, or should everything below also come under the ToC header?